projects/cm_mcu/Tasks.h

```diff
@@ -96,25 +113,74 @@ void MonitorTask(void *parameters);
 #define NFIREFLIES_F1 10
 #define NFIREFLIES_F2 10
 #endif // REV 2
-#define NFIREFLIES (NFIREFLIES_F1 + NFIREFLIES_F2)
+#define NFIREFLIES_IT_F1  6
+#define NFIREFLIES_DAQ_F1 4
+#define NFIREFLIES_IT_F2  6
+#define NFIREFLIES_DAQ_F2 4
+#define CLK_PAGE_COMMAND  1
+#define NFIREFLIES        (NFIREFLIES_F1 + NFIREFLIES_F2)
+
+#define VENDOR_START_BIT_FFDAQ 168
+#define VENDOR_STOP_BIT_FFDAQ  184
+#define VENDOR_START_BIT_FF12  171
+#define VENDOR_STOP_BIT_FF12   187
+
+// pilfered and adapted from http://billauer.co.il/blog/2018/01/c-pmbus-xilinx-fpga-kc705/
+enum pm_type { PM_VOLTAGE,
+  PM_NONVOLTAGE,
+  PM_STATUS,
+  PM_LINEAR11,
+  PM_LINEAR16U,
+  PM_LINEAR16S };
+
+struct dev_moni2c_addr_t {
+  char *name;
+  uint8_t mux_addr; // I2C address of the Mux
+  uint8_t mux_bit;  // port of the mux; write value 0x1U<<mux_bit to the mux register
+  uint8_t dev_addr; // I2C address of device.
+};
 
-void FireFlyTask(void *parameters);
-extern QueueHandle_t xFFlyQueueIn;
-extern QueueHandle_t xFFlyQueueOut;
-SemaphoreHandle_t getFFMutex();
+extern struct dev_moni2c_addr_t ff_moni2c_addrs[NFIREFLIES];
 
-const char *getFFname(const uint8_t i);
-int8_t *test_read(const uint8_t i);
+bool getFFch_low(uint8_t val, int channel);
+bool getFFch_high(uint8_t val, int channel);
 bool isEnabledFF(int ff);
 int8_t getFFtemp(const uint8_t i);
+void getFFpart();
 uint8_t getFFstatus(const uint8_t i);
-bool getFFlos(int i, int channel);
-bool getFFlol(int i, int channel);
-TickType_t getFFupdateTick();
-// FFLY I/O Expander initialization
+int getFFcheckStale();
+TickType_t getFFupdateTick(int ff_t);
 void init_registers_ff();
 
-int disable_xcvr_cdr(const char *name);
+// ff_ctl
+// control the LED
+void LedTask(void *parameters);
+
+#define RED_LED_OFF       (10)
+#define RED_LED_ON        (11)
+#define RED_LED_TOGGLE    (12)
+#define RED_LED_TOGGLE3   (13)
+#define RED_LED_TOGGLE4   (14)
+#define BLUE_LED_OFF      (20)
+#define BLUE_LED_ON       (21)
+#define BLUE_LED_TOGGLE   (22)
+#define BLUE_LED_TOGGLE3  (23)
+#define BLUE_LED_TOGGLE4  (24)
+#define GREEN_LED_OFF     (30)
+#define GREEN_LED_ON      (31)
+#define GREEN_LED_TOGGLE  (32)
+#define GREEN_LED_TOGGLE3 (33)
+#define GREEN_LED_TOGGLE4 (34)
+
+/*
```

Comment:
delete this commented out block of code if not needed

Reply:
cleaned up. Done.